middleware/streammiddleware/logging.go

```diff
@@ -0,0 +1,77 @@
+package streammiddleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/arquivei/foundationkit/errors"
+	"github.com/rs/zerolog/log"
+
+	"github.com/arquivei/goduck"
+)
+
+type streamLogging struct {
+	next goduck.Stream
+}
+
+// WrapWithLogging wraps a @next with a logging that logs when methods are
+// called.
+func WrapWithLogging(next goduck.Stream) goduck.Stream {
+	return streamLogging{
+		next: next,
+	}
+}
+
+func (s streamLogging) Next(ctx context.Context) (response goduck.RawMessage, err error) {
+	const op = errors.Op("streammiddleware.streamLogging.Next")
+	defer func(begin time.Time) {
+		took := time.Since(begin)
+		if err != nil {
+			log.Error().
+				Dur("took", took).
+				Err(errors.E(op, err)).
+				Msg("Error fetching next message")
+		} else {
+			log.Debug().
+				Dur("took", took).
```

Comment:
How about logging the size of the message? it's a cheap information and it could help identify bad messages.